### DISCUSSION TITLE
Ambiguous/multiple categories

### DISCUSSION THREAD

**Discussion by mpadge** [ropensci/statistical-software-review-book#63] [Q&A]
rOpenSci's general software review process requires authors to identify ***one or more*** categories. Statistical software can also potentially fit into multiple categories, but authors are required to document compliance with ***all*** standards from each category. This is a considerable burden, and will pose a particular challenge for people who suspect their package might fit into multiple categories. Could there be any way to allow packages to fit multiple categories, while requiring them to comply with only a subset of all corresponding standards? We currently require compliance with at least half of all category-specific standards. Could this requirement be spread across multiple categories, requiring compliance with at least half of standards, regardless of which category they come from?

(And note that numbers of standards complied with is automatically generated by [the `srr::srr_report()` function](https://docs.ropensci.org/srr/reference/srr_report.html), so those numbers are easy to obtain.)

**Comment by bcjaeger** [comment]
> Could there be any way to allow packages to fit multiple categories while requiring them to comply with only a subset of all corresponding standards?

This is a great question. I think building some conditions into the standards could be helpful, e.g. allowing authors to skip standards related to missing data in the general category if they have met standards regarding missing data in another, such as machine learning. Putting this into practice may be feasible. If authors request standards for the general category plus at least one other category, you can filter the standards presented in the general category (or mark them as optional) based on the other requested categories. 

__Some tangential thoughts__ 

being able to mark some standards as not applicable to your software is very helpful. Checking off the standards for two categories took me a while, but I felt all of the standards that applied to my software helped me improve my documentation and also improve the package itself. I was confused by the following two categories: 

1. machine learning 
2.  regression and supervised learning

My confusion was caused by the decision to put regression into the same category as supervised learning. They certainly do have an overlap (prediction models fitted with regression), but the overlap between ML and supervised learning is much larger so it made me uncertain whether my own software was in the first category or the second.  I wonder if you could turn these two categories into three:

1. Supervised learning
2. Unsupervised learning
3. Statistical inference

I think these three categories have very little overlap, hopefully making it easier for authors to figure out which one their software belongs in. It would not be easy to go back and restructure the standards though, so I completely understand if you choose not to act on this suggestion. Hopefully it is helpful!

**Reply by mpadge** [reply]
Thanks @bcjaeger, glad to hear that you found that the process helped you improve both documentation and code! I also appreciate your thoughts on category confusion. Clarifying categories was a very difficult exercise. That said, note that we approached it as empirically as we possibly could, with much of the raw data in [our interactive network of statistical terms](https://ropenscilabs.github.io/statistical-software/abstracts/network-terms/index.html). "Inference" is one of the terms there, but you'll see that it's not very central, and not sufficiently inter-connected to have become a category in its own right. (Easiest way to find it is to locate "*dimensionality reduction*", and then look for peripheral connections both to "inference" and adjacent "effect sizes".) That graph also clarifies why "*machine learning*" is its own category (called "ML" there - pull the dot to see how distinct and connected it is).

We are expecting the definitions to evolve over time, but note at least at this early stage that any potential modification of categories would have to reflect the empirical research represented in that diagram. We'll collate information on ambiguous categorisations as we review more pacakges, starting with yours. Eventually we may collate enough internal empirical evidence from package reviews to reconsider categories, at which time we'll definitely ask for more input from you. Thanks, and feel free to offer any other feedback in the meantime.

**Reply by bcjaeger** [reply]
No problem at all, thanks for considering my thoughts! Nobody could argue against using data to define the main categories. I understand your stance there. 

I think it is worth pursuing your original thought about allowing packages to fit multiple categories while only requiring compliance with a subset of standards in those categories. Figuring out the best way to do it is challenging. I will report back to you if I have any thoughts that may be helpful.
